index.js

```diff
@@ -72,9 +72,11 @@ Storage.prototype._end = function (opts, cb) {
   next(0)
 
   function next (i) {
-    if (!self.stores[i]) return cb()
+    if (i === self.stores.length) return cb()
+    if (!self.stores[i]) return next(i + 1)
     self.stores[i].storage.end(opts, function (err) {
       if (err) return cb(err)
+      delete self.stores[i]
```

Comment:
i had an implementation with `.shift()` but went with this because i expect it to be more performant. it's your call though, how important performance is for this code path.